src/Core/Waiter.php

```diff
@@ -197,7 +198,7 @@ public function __destruct()
             if ($delay > $timeout - (\microtime(true) - $start)) {
                 break;
             }
-            \usleep((int) ceil($delay * 1000000));
+            \usleep((int) ceil($delay * 1000000)); // This looks suspicious, there should be never be a need to sleep to me
```

Comment:
In AWS, a waiter implements a polling pattern:
- sqs->createQueue()
- sqs->queueExist()->wait()
- - sqs->getQueueUrl() // 404
- - sleep 5sec
- - sqs->getQueueUrl() // 404
- - sleep 5sec
- - sqs->getQueueUrl() // 200

$delay, is the time to wait between 2 calls

Reply:
there is a timeout argument already - it's enough to "wait" for me.

Reply:
I think you missed a piece here. This method is polling an endpoint.

`sqs->queueExist()->wait(300, 5)` means, call the endpoint `getQueueUrl` every 5 seconds until the response is not 404 (the methods `getQueueUrl` responds in few milliseconds). Abort after 5 minutes.

Not sleeping between each call, will triggers a RateLimit error from AWS.

Reply:
OK, you must be right :)